src/containers/shared/components/Currency.tsx

```diff
@@ -40,12 +40,19 @@ const Currency = (props: Props) => {
       display
     )
   } else {
-    const currencyCode =
+    let currencyCode =
       currency?.length === NON_STANDARD_CODE_LENGTH &&
       currency?.substring(0, 2) !== LP_TOKEN_IDENTIFIER
         ? hexToString(currency)
         : currency
 
+    if (
+      currency?.length === NON_STANDARD_CODE_LENGTH &&
+      currencyCode.length <= 3
```

Comment:
```suggestion
      currencyCode.length === 3
```
I don't feel strongly about it, but I think prepending `Fake` only when `currencyCode.length === 3` is fine.